integration_test/integration_test.go

```diff
@@ -298,6 +372,621 @@ func (t *WASMIntegrationTestSuite) Test04_ContractExecution() {
+				)
+
+				err := applyVoteTallyingProposal(
+					desc.GetConnectionWithContext(context.Background()),
+					proposalContent,
+					t.ZeroRewardValidatorWallet2,
+					[]*WalletInfo{t.ValidatorWallet1, t.ValidatorWallet2, t.ValidatorWallet3, t.ValidatorWallet4},
+				)
+
+				if err != nil {
+					return err
+				}
+
+				return nil
+			})
+		}
+
+		err := eg.Wait()
+
+		if assert.NoError(t.T(), err) {
+			fmt.Println("Proposal successfully applied!")
+		} else {
+			fmt.Println("Error detected on the proposal")
+			t.T().Fail()
+		}
+	}
+
+	fmt.Println("Waiting 30sec for the proposal passing...")
+	time.Sleep(time.Second * 30)
+
+	{
+		fmt.Println("Check existence of the zero reward validator")
+
+		client := zrValidatorType.NewQueryClient(desc.GetConnectionWithContext(context.Background()))
+		validatorStatus, err := client.ZeroRewardValidators(context.Background(), &zrValidatorType.QueryZeroRewardValidatorsRequest{})
+		assert.NoError(t.T(), err)
+
+		thisVoluntaryValAddress := sdktypes.ValAddress(t.ZeroRewardValidatorWallet2.ByteAddress).String()
+
+		if assert.NotContains(t.T(), validatorStatus.GetZeroRewardValidators(), thisVoluntaryValAddress) {
+			fmt.Println(thisVoluntaryValAddress, "successfully removed from the validator set!")
+		} else {
+			fmt.Println(thisVoluntaryValAddress, "still exist!")
+			t.T().Fail()
+		}
+	}
+}
+
+func (t *WASMIntegrationTestSuite) Test14_TryChangingGeneralValidatorToZeroRewardValidator_ShouldFail() {
+	amt := sdktypes.NewInt(1_000000_000000_000000)
+
+	{
+		fmt.Println("Try registering as a zero reward validator from the general validator...")
+
+		proposalContent, err := zrValidatorType.NewRegisterZeroRewardValidatorProposal(
+			"register_zeroreward_validator_from_general_validator",
+			"Test zero reward validator registary",
+			sdktypes.AccAddress(t.ValidatorWallet1.ByteAddress.Bytes()),
+			sdktypes.ValAddress(t.ValidatorWallet1.ByteAddress.Bytes()),
+			&ed25519.PubKey{Key: t.Validator1PVKey.PubKey.Bytes()},
+			sdktypes.NewCoin("axpla", amt), // smaller amount than other basic validators
+			stakingtype.NewDescription("zeroreward_validator", "", "", "", ""),
+		)
+
+		assert.NoError(t.T(), err)
+
+		err = applyVoteTallyingProposal(
+			desc.GetConnectionWithContext(context.Background()),
+			proposalContent,
+			t.ZeroRewardValidatorWallet2,
+			[]*WalletInfo{t.ValidatorWallet1, t.ValidatorWallet2, t.ValidatorWallet3, t.ValidatorWallet4},
+		)
+
+		if assert.Error(t.T(), err) {
+			fmt.Println(err)
+			fmt.Println("Expected failure! Test succeeded!")
+		} else {
+			fmt.Println("Unexpected situation. Failure")
+			t.T().Fail()
+		}
+	}
+}
+
+func (t *WASMIntegrationTestSuite) Test15_ValidatorActiveSetChange() {
+	amt := sdktypes.NewInt(1_200000_000000_000000)
+	var maxValidators uint32 = 5
+
+	expectedChange := []paramtype.ParamChange{
+		paramtype.NewParamChange("staking", "MaxValidators", fmt.Sprintf("%d", maxValidators)),
```

Comment:
I think it's better 
`"staking"` -> `stakingtypes.ModuleName`
`"MaxValidators"` -> `stakingtypes.KeyMaxValidators`

Reply:
https://github.com/xpladev/xpla/pull/84/commits/970f120042183ba51ec1167ec3be86bdf469e179